velox/common/caching/SsdFile.cpp

```diff
@@ -690,30 +673,33 @@ void SsdFile::deleteCheckpoint(bool keepLog) {
   if (checkpointDeleted_) {
     return;
   }
-  if (evictLogFd_ >= 0) {
-    if (keepLog) {
-      ::lseek(evictLogFd_, 0, SEEK_SET);
-      ::ftruncate(evictLogFd_, 0);
-      ::fsync(evictLogFd_);
-    } else {
-      ::close(evictLogFd_);
-      evictLogFd_ = -1;
+
+  if (evictLogWriteFile_ != nullptr) {
+    try {
+      if (keepLog) {
+        evictLogWriteFile_->truncate(0);
+        evictLogWriteFile_->flush();
+      } else {
+        evictLogWriteFile_->close();
+        fs_->remove(getEvictLogFilePath());
+        evictLogWriteFile_.reset();
+      }
+    } catch (const std::exception& e) {
+      ++stats_.deleteCheckpointErrors;
+      VELOX_SSD_CACHE_LOG(ERROR) << "Error in deleting evictLog: " << e.what();
     }
   }
 
   checkpointDeleted_ = true;
   const auto logPath = getEvictLogFilePath();
-  int32_t logRc = 0;
-  if (!keepLog) {
-    logRc = ::unlink(logPath.c_str());
-  }
   const auto checkpointPath = getCheckpointFilePath();
   const auto checkpointRc = ::unlink(checkpointPath.c_str());
-  if ((logRc != 0) || (checkpointRc != 0)) {
-    ++stats_.deleteCheckpointErrors;
+  if (checkpointRc != 0) {
     VELOX_SSD_CACHE_LOG(ERROR)
-        << "Error in deleting log and checkpoint. log: " << logRc
-        << " checkpoint: " << checkpointRc;
+        << "Error in deleting checkpoint: " << checkpointRc;
+  }
+  if ((evictLogWriteFile_ != nullptr) || (checkpointRc != 0)) {
```

Comment:
We don't need to check evictLogWriteFile_ here as we have already increment in the try catch above for log eviction error? Also could add unit test with fault injection error to see if the checkpoint error stats is updated properly? Thanks!